### PR TITLE
Added plugin version print to stack trace

### LINKED ISSUE #876: Feature request: plugin version
Add please plugin version print on plugin stack trace. It can help developers to support their plugins and quickly found corrupted section in plugin by knewing a version.

Example:
```
L 08/26/2018 - 14:43:21: [SM] Exception reported: Client index 1 is a bot
L 08/26/2018 - 14:43:21: [SM] Blaming: testplugin.smx (Version 1.1.1)
L 08/26/2018 - 14:43:21: [SM] Call stack trace:
L 08/26/2018 - 14:43:21: [SM]   [0] ThrowNativeError
....
```

### PR DESCRIPTION
**Pull request by TiBarification**
This change refers and closes #876

### CODE REVIEW

File: core/logic/DebugReporter.cpp [hunk 1]
```diff
@@ -177,6 +182,7 @@ void DebugReport::ReportError(const IErrorReport &report, IFrameIterator &iter)
 				if (plugin)
 				{
 					blame = plugin->GetFilename();
+					plversion = plugin->GetPublicInfo()->version;
 				} else {
 					blame = iter.Context()->GetRuntime()->GetFilename();
```
Comment (TiBarification) [inline core/logic/DebugReporter.cpp]: Is there any way to get version in runtime?

**Comment by Bara** [comment]
That would pretty useful. 👍

File: core/logic/DebugReporter.cpp [hunk 2]
```diff
@@ -68,14 +68,16 @@ void DebugReport::GenerateErrorVA(IPluginContext *ctx, cell_t func_idx, int err,
...
-	const char *plname = pluginsys->FindPluginByContext(ctx->GetContext())->GetFilename();
+	IPlugin* plugin = pluginsys->FindPluginByContext(ctx->GetContext());
+	const char *plname = plugin->GetFilename();
+	const char *plversion = plugin->GetPublicInfo()->version;
 	const char *error = g_pSourcePawn2->GetErrorString(err);
 
 	if (error)
 	{
-		g_Logger.LogError("[SM] Plugin \"%s\" encountered error %d: %s", plname, err, error);
+		g_Logger.LogError("[SM] Plugin \"%s\" (Version: %s) encountered error %d: %s", plname, (*plversion != '\0') ? plversion : "Not specified", err, error);
```
Comment (KyleSanderson) [inline core/logic/DebugReporter.cpp]: IS_STR_FILLED; check other usages.

File: core/logic/DebugReporter.cpp [hunk 3]
```diff
@@ -103,14 +105,16 @@ void DebugReport::GenerateCodeError(IPluginContext *pContext, uint32_t code_addr
...
-	const char *plname = pluginsys->FindPluginByContext(pContext->GetContext())->GetFilename();
+	IPlugin* plugin = pluginsys->FindPluginByContext(pContext->GetContext());
+	const char *plname = plugin->GetFilename();
+	const char *plversion = plugin->GetPublicInfo()->version;
 	const char *error = g_pSourcePawn2->GetErrorString(err);
 
 	if (error)
 	{
-		g_Logger.LogError("[SM] Plugin \"%s\" encountered error %d: %s", plname, err, error);
+		g_Logger.LogError("[SM] Plugin \"%s\" (Version: %s) encountered error %d: %s", plname, (*plversion != '\0') ? plversion : "Not specified", err, error);
```
Comment (KyleSanderson) [inline core/logic/DebugReporter.cpp]: Get rid of the ternary, assign the const to the actual variable.

**Pull request closed without merging**